pandas/core/arrays/categorical.py

```diff
@@ -429,7 +429,7 @@ def ordered(self):
         return self.dtype.ordered
 
     @property
-    def dtype(self):
+    def dtype(self) -> 'CategoricalDtype':
```

Comment:
Do you need the forward reference here or can you explicitly mention the object?

Reply:
It's always tricky to use non-forward references to types that are in your own code base. So right now I think the forward reference is *not* needed, BUT if we at a later stage do an import from categorical.py (this file) and afterwards import from dtypes.py (the file where ``CategoricalDtype`` is defined, we might get circular import errors that tracks back to this type hint not being forward referenced.

This is solved in Python 3.7 through [PEP 563](https://www.python.org/dev/peps/pep-0563/), but won't help Pandas for some while.

So my opinion is that Pandas types should be forward referenced, for sanity's sake, untill we drop 3.6.

Reply:
this is not needed, pls use the actual type.